lib/FetchEnvs.js

```diff
@@ -181,10 +181,14 @@ module.exports = class FetchEnvs {
   #processEnvFrom(envFrom) {
     return Promise.all(envFrom.map((element) => {
       const { configMapRef, secretMapRef, genericMapRef } = element;
+
+      if (!configMapRef && !secretMapRef && !genericMapRef) {
+        throw new Error(`oneOf configMapRef, secretMapRef, genericMapRef must be defined. Got: ${JSON.stringify(element)}`);
+      }
+
       if (configMapRef) return this.#configMapRef(element);
-      if (secretMapRef) return this.#secretMapRef(element);
-      if (genericMapRef) return this.#genericMapRef(element);
-      return element;
+      else if (secretMapRef) return this.#secretMapRef(element);
```

Comment:
The `else`s are not necessary with the `return`.  if the if condition passes, it terminates the function. if it doesn't, it moves on to the next

you could probably just omit `if (genericMapRef)` from the last line

```javascript
if (configMapRef) return this.#configMapRef(element);
if (secretMapRef) return this.#secretMapRef(element);
return this.#genericMapRef(element)
```